lint/mypy.bzl

```diff
@@ -0,0 +1,190 @@
+"""API for declaring a mypy lint aspect that visits py_library rules.
+
+Typical usage:
+
+```
+load("@aspect_rules_lint//lint:mypy.bzl", "mypy_aspect")
+
+mypy = mypy_aspect(
+    binary = "@@//:mypy",
+    configs = "@@//:pyproject.toml",
+)
+```
+
+This code inspired from https://github.com/bazel-contrib/bazel-mypy-integration
+with appreciation to [Jonathon Belotti](https://github.com/thundergolfer)
+and the other [contributors](https://github.com/bazel-contrib/bazel-mypy-integration/graphs/contributors)
+especially [David Zbarsky](https://github.com/dzbarsky)
+
+### TODO
+
+- Document --norun_validations
+- Support third-party definitions from stubs installed from type-* packages
+- Allow configured typeshed repo, e.g. args.add("--custom-typeshed-dir", "external/my_typeshed")
+- Set MYPYPATH if needed for cases where 1p sources don't resolve correctly to transitive .py sources: https://github.com/dzbarsky/mypy-bazel-experiments/blob/master/build/py/mypy.bzl#L163
```

Comment:
fyi this repo isn't public, but if this is just notes for you that's fine :)

Reply:
yeah it's meant for my own notes, I could move out of the block that's docgen'ed